datasetinsights/estimators/faster_rcnn.py

```diff
@@ -192,12 +197,14 @@ def train(self, train_data, val_data=None, **kwargs):
         val_loader = dataloader_creator(
             config, val_dataset, val_sampler, VAL, self.distributed
         )
-        self.train_loop(
-            train_dataloader=train_loader,
-            label_mappings=label_mappings,
-            val_dataloader=val_loader,
-            train_sampler=train_sampler,
-        )
+        with self.mltracking.start_run(run_name=const.DEFAULT_RUN_NAME):
```

Comment:
- why doesn't the run start when the mlflow tracker instance is created?
- why does this ignore the run name that may be passed by the user in a config or CLI flag? this part in particular should include a unit test

Reply:
mlflow run starts when your experiment run starts and it ends when your experiment run end.
 
There are two ways to start mlflow run.
1. Using "with" block: mlflow ends the runs automatically when you exit the with block.
2. using manual start and manual end
Documentation: https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.start_run

- I can start the run when tracker instance gets created, this will allow to change run name from CLI. But I need to put train/eval run under try block and end mlflow run in catch/finally block and change the status of run according to fail or success.